nvdaHelper/localWin10/oneCoreSpeech.cpp

```diff
@@ -30,6 +37,130 @@ using namespace winrt::Windows::Media;
 using namespace winrt::Windows::Foundation::Collections;
 using winrt::Windows::Foundation::Metadata::ApiInformation;
 
+std::recursive_mutex InstanceManager::_instanceStateMutex;
+std::vector<OcSpeech*> InstanceManager::_terminatedInstances;
+std::unique_ptr<OcSpeech> InstanceManager::_instance;
+std::atomic<InstanceState> InstanceManager::_instanceState = InstanceState::notInitialized;
+std::condition_variable_any InstanceManager::_readyForInitialization;
+std::atomic_int SpeakThreadGuard::_speechThreads = 0;
+
+SpeakThreadGuard::SpeakThreadGuard() {
+	/*
+	When initialized, increases the count of active speech threads
+	waiting on a callback.
+	When exiting the scope of initialization, decreases the count
+	of active speech threads and deletes the OneCore instance if it has
+	been terminated and no speech callbacks are waiting.
+	*/

```

Comment:
C++ docs (like python doc strings) are typically written on the line before the declaration, rather than within the definition.

It is also helpful to have reasoning for the implementation within the definition.